src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -380,6 +378,23 @@ def is_shared_gallery_image_id(image_reference):
     return False
 
 
+_VM_IMAGE_ID_PATTERN = r'^/subscriptions/[^/]*/resourceGroups/[^/]*/providers/Microsoft.Compute/virtualMachines/.*$'
+_IMAGE_VERSION_ID_PATTERN = r'^/subscriptions/[^/]*/resourceGroups/[^/]*/providers/Microsoft.Compute/galleries/' \
+                            r'[^/]*/images/[^/]*/versions/.*$'
+
+
+def is_valid_image_id(image_id, is_vm):
+    if not image_id:
+        return True
+
+    image_id_pattern = re.compile(_VM_IMAGE_ID_PATTERN if is_vm else _IMAGE_VERSION_ID_PATTERN, re.IGNORECASE)
+
+    if image_id_pattern.match(image_id):
+        return True
+
+    return False
```

Comment:
It is suggested to split this method into two decoupling methods (such as `is_valid_image_version_id()` and `is_valid_vm_resource_id()`, so that it can be reused when there are similar requirements in the future